app/controllers/posts_controller.rb

```diff
@@ -1,16 +1,31 @@
 class PostsController < ApplicationController
   def index
-    @user = User.find(params[:user_id])
-    @posts = Post.where(author_id: params[:user_id])
-    @comments = Comment.where(post_id: params[:post_id])
-    @likes = Like.where(post_id: params[:post_id])
+    @user = User.includes(posts: %i[comments likes]).find(params[:user_id])
+    @posts = @user.posts
+    @comments = Comment.where(post_id: @posts.pluck(:id)).group_by(&:post_id)
```

Comment:
- [ ] To solve the  N+1 problem when fetching all posts and their comments for a user I think 🤔 It will be much better to place the include on the posts and not the user, since we only have a single user that should fetch all the posts Your code may look similar to this one:

`@posts = @user.posts.includes(:comments)`